lib/extensions/postgres_cdc_rls/replication_poller.ex

```diff
@@ -220,6 +152,93 @@ defmodule Extensions.PostgresCdcRls.ReplicationPoller do
     {:noreply, prepare_replication(state)}
   end
 
+  def slot_name_suffix() do
+    case System.get_env("SLOT_NAME_SUFFIX") do
+      nil ->
+        ""
+
+      value ->
+        Logger.debug("Using slot name suffix: " <> value)
+        "_" <> value
+    end
+  end
+
+  defp convert_errors([_ | _] = errors), do: errors
+
+  defp convert_errors(_), do: nil
+
+  defp connect_db(host, port, name, user, pass, socket_opts, ssl_enforced) do
+    {host, port, name, user, pass} = decrypt_creds(host, port, name, user, pass)
+
+    [
+      hostname: host,
+      port: port,
+      database: name,
+      password: pass,
+      username: user,
+      queue_target: @queue_target,
+      parameters: [application_name: "realtime_rls"],
+      socket_options: socket_opts
+    ]
+    |> maybe_enforce_ssl_config(ssl_enforced)
+    |> Postgrex.start_link()
+  end
+
+  defp prepare_replication(
+         %{backoff: backoff, conn: conn, slot_name: slot_name, retry_count: retry_count} = state
+       ) do
+    case Replications.prepare_replication(conn, slot_name) do
+      {:ok, _} ->
+        send(self(), :poll)
+        state
+
+      {:error, error} ->
+        Logger.error("Prepare replication error: #{inspect(error)}")
+        {timeout, backoff} = Backoff.backoff(backoff)
+        retry_ref = Process.send_after(self(), :retry, timeout)
+        %{state | backoff: backoff, retry_ref: retry_ref, retry_count: retry_count + 1}
+    end
+  end
+
+  defp record_list_changes_telemetry(time, tenant) do
+    Realtime.Telemetry.execute(
+      [:realtime, :replication, :poller, :query, :stop],
+      %{duration: time},
+      %{tenant: tenant}
+    )
+  end
+
+  defp handle_list_changes_result(
+         {:ok,
+          %Postgrex.Result{
+            columns: ["wal", "is_rls_enabled", "subscription_ids", "errors"] = columns,
+            rows: [_ | _] = rows,
+            num_rows: rows_count
+          }},
+         tenant
+       ) do
+    rows
+    |> Enum.reduce([], fn row, acc ->
+      columns
+      |> Enum.zip(row)
+      |> generate_record()
+      |> then(fn
+        nil -> acc
+        record_struct -> [record_struct | acc]
+      end)
+    end)
+    |> Enum.reverse()
+    |> Enum.each(fn change ->
+      topic = "realtime:postgres:" <> tenant
+      Phoenix.PubSub.broadcast_from(PubSub, self(), topic, change, MessageDispatcher)
+    end)
```

Comment:
You can even wrap the whole thing a comprehension if you want:

```suggestion
    for row <- rows,
        change <- columns |> Enum.zip(row) |> generate_record() |> List.wrap() do
      topic = "realtime:postgres:" <> tenant
      Phoenix.PubSub.broadcast_from(PubSub, self(), topic, change, MessageDispatcher)
    end)
```